enterprise/tailnet/coordinator_test.go

```diff
@@ -0,0 +1,260 @@
+package tailnet_test
+
+import (
+	"net"
+	"testing"
+
+	"github.com/google/uuid"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"cdr.dev/slog/sloggers/slogtest"
+
+	"github.com/coder/coder/coderd/database"
+	"github.com/coder/coder/enterprise/tailnet"
+	agpl "github.com/coder/coder/tailnet"
+	"github.com/coder/coder/testutil"
+)
+
+func TestCoordinatorSingle(t *testing.T) {
```

Comment:
I think writing a benchmark would be helpful before merge. @johnstcn ran into some oddities with PostgreSQL pubsub performance, so we should ensure those limits aren't applicable here.

Reply:
Since this already works fine for v1 I think it would be good to find the limits for this later and recommend scaling in the docs.